src/base/bittorrent/session.cpp

```diff
@@ -321,6 +321,11 @@ QString Session::tempPath() const
     return m_tempPath;
 }
 
+QString Session::torrentTempPath(const InfoHash &hash) const
+{
+    return m_tempPath + QString(hash) + "/";
```

Comment:
>We can do, for example, as Git does - only use the first 7 characters of hash. 

This should work.
then you can just substring the sha-1 of the torrent file instead of hashing it again.